engine/graphics/src/graphics.cpp

```diff
@@ -1161,7 +1412,22 @@ namespace dmGraphics
     {
         dmPlatform::SetSwapInterval(g_functions.m_GetWindow(context), swap_interval);
     }
+
+    ///////////////////////////////////////////////////
+    //////// BASE ADAPTER SPECIFIC FUNCTIONS //////////
+    uint32_t GetUniformCount(HProgram prog)
+    {
+        Program* p = (Program*) prog;
+        return p->m_Uniforms.Size();
+    }
+    void GetUniform(HProgram prog, uint32_t index, Uniform* uniform_desc)
+    {
+        Program* p = (Program*) prog;
+        *uniform_desc = p->m_Uniforms[index];
+    }
```

Comment:
This is a somewhat new pattern for the graphics layer, where where we use a "base struct" that is shared for all adapters:

struct BaseThing
{
   int m_Shared;
};

struct OpenGLThing
{
  BaseThing m_Base;
  OpenGLThing m_Thing;
};

struct VulkanThing
{
  BaseThing m_Base;
  VulkanThing m_Thing;
};